pkg/schedulingtypes/replicascheduler.go

```diff
@@ -108,6 +108,38 @@ func (s *ReplicaScheduler) Kind() string {
 	return RSPKind
 }
 
+func (s *ReplicaScheduler) RegisterPlugins(kind string, apiResource *metav1.APIResource, stopChan <-chan struct{}) {
+	var adapter adapters.Adapter
+	switch kind {
+	case FederatedDeployment:
+		adapter = adapters.NewFederatedDeploymentAdapter(s.fedClient)
+	case FederatedReplicaSet:
+		adapter = adapters.NewFederatedReplicaSetAdapter(s.fedClient)
+	default:
+		return
+	}
+
+	if _, ok := s.plugins[kind]; ok {
+		return
+	}
+
+	glog.Infof("Kind %s is registered to the scheduler plugin", kind)
+	s.plugins[kind] = NewPlugin(
+		adapter,
+		apiResource,
+		s.fedClient,
+		s.pluginArgs.kubeClient,
+		s.pluginArgs.crClient,
+		s.pluginArgs.fedNamespace,
+		s.pluginArgs.clusterNamespace,
+		s.pluginArgs.targetNamespace,
+		s.pluginArgs.federationEventHandler,
+		s.pluginArgs.clusterEventHandler,
+		s.pluginArgs.handlers,
+	)
+	s.plugins[kind].Start(stopChan)
```

Comment:
Given that this PR ensures that plugins are only enabled by calling this method, the `Start` method of the controller no longer needs to attempt to start plugins.